controllers/mirrorpeer_controller.go

```diff
@@ -140,10 +141,18 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	if err != nil {
 		logger.Info("Error while exchanging tokens", "MirrorPeer", mirrorPeer)
 		mirrorPeer.Status.Phase = multiclusterv1alpha1.ExchangingSecret
+		if k8serrors.IsNotFound(err) {
+			logger.Info("Secrets not found; Attempting to reconcile in 5 seconds")
+			statusErr := r.Client.Status().Update(ctx, &mirrorPeer)
+			if statusErr != nil {
+				logger.Error(statusErr, "Error occurred while updating the status of mirrorpeer", "MirrorPeer", mirrorPeer)
+			}
+			return ctrl.Result{RequeueAfter: time.Second * 5}, nil
+		}
```

Comment:
Instead of this, can we do something like this?

When MirrorPeer is created (i.e. Status.Phase == nil), immediately set it to `ExchangingSecret`.
Then `checkTokenExchangeStatus`. If token is exchanged, update status to `ExchangedSecret` and immediately reconcile if it fails (don't wait for 5s). If token is not exchanged, don't do anything.

Reply:
Do not change error handling for main controller logic.

Reply:
> When MirrorPeer is created (i.e. Status.Phase == nil), immediately set it to ExchangingSecret

Done 

> If token is exchanged, update status to ExchangedSecret and immediately reconcile if it fails (don't wait for 5s). If token is not exchanged, don't do anything.

Changed the reconcile timing to 0 and it will be done immediately.